scripts/core/ui/components/select2.tsx

```diff
@@ -14,14 +14,15 @@ interface IProps<T> {
     renderItem(item: T): JSX.Element;
     getItemValue(item: T): string;
     onSelect(value: string): void;
-    onSearch?(search: string): void;
-    onFocus: boolean;
+    onSearch?(search: string): Promise<any>;
+    onFocus?: boolean | {hideOptions: boolean};
```

Comment:
imo it can use some renaming, we usually have callbacks named `onSomething` and here it's bool or dict, and I'm not exactly sure what they do